src/resources/addys/claimables/query.ts

```diff
@@ -0,0 +1,85 @@
+import { NativeCurrencyKey } from '@/entities';
+import { RainbowFetchClient } from '@/rainbow-fetch';
+import { QueryConfigWithSelect, QueryFunctionArgs, QueryFunctionResult, createQueryKey } from '@/react-query';
+import { SUPPORTED_CHAIN_IDS } from '@/references';
+import { useQuery } from '@tanstack/react-query';
+import { ADDYS_API_KEY } from 'react-native-dotenv';
+import { ConsolidatedClaimablesResponse } from './types';
+import { logger, RainbowError } from '@/logger';
+import { parseClaimables } from './utils';
+import { useRemoteConfig } from '@/model/remoteConfig';
+import { CLAIMABLES, useExperimentalFlag } from '@/config';
+import { useConnectedToHardhatStore } from '@/state/connectedToHardhat';
+import { IS_TEST } from '@/env';
+
+const addysHttp = new RainbowFetchClient({
+  baseURL: 'https://addys.p.rainbow.me/v3',
+  headers: {
+    Authorization: `Bearer ${ADDYS_API_KEY}`,
+  },
+});
+
+// ///////////////////////////////////////////////
+// Query Types
+
+export type ClaimablesArgs = {
+  address: string;
+  currency: NativeCurrencyKey;
+  testnetMode?: boolean;
+};
+
+// ///////////////////////////////////////////////
+// Query Key
+
+export const claimablesQueryKey = ({ address, currency, testnetMode }: ClaimablesArgs) =>
+  createQueryKey('claimables', { address, currency, testnetMode }, { persisterVersion: 1 });
+
+type ClaimablesQueryKey = ReturnType<typeof claimablesQueryKey>;
+
+// ///////////////////////////////////////////////
+// Query Function
+
+async function claimablesQueryFunction({ queryKey: [{ address, currency, testnetMode }] }: QueryFunctionArgs<typeof claimablesQueryKey>) {
+  try {
+    const url = `/${SUPPORTED_CHAIN_IDS({ testnetMode }).join(',')}/${address}/claimables`;
+    const { data } = await addysHttp.get<ConsolidatedClaimablesResponse>(url, {
+      params: {
+        currency: currency.toLowerCase(),
+      },
+      timeout: 20000,
+    });
+
+    if (data.metadata.status !== 'ok') {
+      logger.error(new RainbowError('[userAssetsQueryFunction]: Failed to fetch user assets (API error)'), {
+        message: data.metadata.errors,
+      });
+    }
+
+    return parseClaimables(data.payload.claimables, currency);
+  } catch (e) {
+    logger.error(new RainbowError('[userAssetsQueryFunction]: Failed to fetch user assets (client error)'), {
+      message: (e as Error)?.message,
+    });
+  }
```

Comment:
when `data.metadata.status !== 'ok'` will we have a `data.payload.claimables`? just to be sure it wont log like `(API error)` and then right after `(client error)` because it couldn't access `payload.claimables`

Reply:
good question, i believe `data.payload.claimables` should always be at least an empty array, according to the addys backend types